### PR TITLE
Feat/add versioning components

### PR DESCRIPTION
**Pull request by russojrv-zz**
**What has changed?**

- We have implemented a new feature in the component-repository where we can have virtual components one level above the existing components and group them. Consequently, we can have multiple versions of one component.
- We added a new configuration collection where (for now) we can store one authClientId for tenant. In the future, we could add more configuration per component/tenant in this collection.

**Does a specific change require especially careful review?**

_It would be nice if you can take a look to the permissions and querys_

**What is still open or a known issue?**

This may be related to this issue: [https://github.com/openintegrationhub/openintegrationhub/issues/1174](https://github.com/openintegrationhub/openintegrationhub/issues/1174)

### CODE REVIEW

File: lib/component-repository/package.json [hunk 1]
```diff
@@ -28,11 +28,12 @@
     "eslint": "7.32.0",
     "eslint-plugin-mocha": "9.0.0",
     "eslint-plugin-node": "11.1.0",
+    "faker": "^5.5.3",
```
Comment (heggert) [inline lib/component-repository/package.json]: Please use this lib instead:
https://www.npmjs.com/package/@faker-js/faker

Why?
https://fossa.com/blog/npm-packages-colors-faker-corrupted/

File: lib/component-repository/src/routes/virtual-components/index.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const bodyParser = require('body-parser').json();
```
Comment (heggert) [inline lib/component-repository/src/routes/virtual-components/index.js]: I would use the native functionality of express and do without the separate `bodyParser`. Basically, you only need to replace one line.
https://dev.to/taylorbeeston/you-probably-don-t-need-body-parser-in-your-express-apps-3nio

**Review by heggert — CHANGES_REQUESTED** [review]

File: lib/component-repository/package.json (hunk 1 above)
Comment (russojrv-zz) [reply lib/component-repository/package.json]: Replaced!

File: lib/component-repository/src/routes/virtual-components/index.js (hunk 2 above)
Comment (russojrv-zz) [reply lib/component-repository/src/routes/virtual-components/index.js]: Replaced to the native express functionality.

**Review by heggert — APPROVED** [review]

**Pull request merged**